pkg/scheduler/plugins/deviceshare/utils.go

```diff
@@ -0,0 +1,238 @@
+}
+
+func convertCommonDeviceResource(podRequest corev1.ResourceList, deviceType schedulingv1alpha1.DeviceType) corev1.ResourceList {
+	if podRequest == nil || len(podRequest) == 0 {
+		klog.Warningf("pod request should not be empty")
+		return nil
+	}
+	var resources corev1.ResourceList
+	switch deviceType {
+	case schedulingv1alpha1.RDMA:
+		if value, ok := podRequest[apiext.KoordRDMA]; ok {
+			resources = corev1.ResourceList{
+				apiext.KoordRDMA: value,
+			}
+		}
+	case schedulingv1alpha1.FPGA:
+		if value, ok := podRequest[apiext.KoordFPGA]; ok {
+			resources = corev1.ResourceList{
+				apiext.KoordFPGA: value,
+			}
+		}
+	default:
+		klog.Warningf("device type %v is not supported yet", deviceType)
+		return nil
+	}
+	return resources
+}
+
+// nvidia.com/gpu means applying for full-card
+// koordinator.sh/gpu means applying for cards in percentile
+// convertGPUResource will convert either nvidia.com/gpu or koordinator.sh/gpu to koordinator.sh/gpu-core and koordinator.sh/gpu-memory-ratio
+func convertGPUResource(podRequest corev1.ResourceList, combination uint) corev1.ResourceList {
+	if podRequest == nil || len(podRequest) == 0 {
+		klog.Warningf("pod request should not be empty")
+		return nil
+	}
+	switch combination {
+	case gpuCoreExist | gpuMemoryExist:
+		return corev1.ResourceList{
+			apiext.GPUCore:   podRequest[apiext.GPUCore],
+			apiext.GPUMemory: podRequest[apiext.GPUMemory],
+		}
+	case gpuCoreExist | gpuMemoryRatioExist:
+		return corev1.ResourceList{
+			apiext.GPUCore:        podRequest[apiext.GPUCore],
+			apiext.GPUMemoryRatio: podRequest[apiext.GPUMemoryRatio],
+		}
+	case koordGpuExist:
+		return corev1.ResourceList{
+			apiext.GPUCore:        podRequest[apiext.KoordGPU],
+			apiext.GPUMemoryRatio: podRequest[apiext.KoordGPU],
+		}
+	case nvidiaGpuExist:
+		nvidiaGpu := podRequest[apiext.NvidiaGPU]
+		return corev1.ResourceList{
+			apiext.GPUCore:        *resource.NewQuantity(nvidiaGpu.Value()*100, resource.DecimalSI),
+			apiext.GPUMemoryRatio: *resource.NewQuantity(nvidiaGpu.Value()*100, resource.DecimalSI),
+		}
+	}
+	return nil
+}
+
+func isMultipleCommonDevicePod(podRequest corev1.ResourceList, deviceType schedulingv1alpha1.DeviceType) bool {
+	if podRequest == nil || len(podRequest) == 0 {
+		klog.Warningf("pod request should not be empty")
+		return false
+	}
+	switch deviceType {
+	case schedulingv1alpha1.RDMA:
+		rdma := podRequest[apiext.KoordRDMA]
+		return rdma.Value() > 100 && rdma.Value()%100 == 0
+	case schedulingv1alpha1.FPGA:
+		fpga := podRequest[apiext.KoordFPGA]
+		return fpga.Value() > 100 && fpga.Value()%100 == 0
+	default:
+		return false
+	}
+}
+
+func isMultipleGPUPod(podRequest corev1.ResourceList) bool {
+	if podRequest == nil || len(podRequest) == 0 {
+		klog.Warningf("pod request should not be empty")
+		return false
+	}
+	gpuCore := podRequest[apiext.GPUCore]
+	return gpuCore.Value() > 100 && gpuCore.Value()%100 == 0
```

Comment:
for gpuCore.value == 350, is it  invalid ? or we just treat it as single card request?

Reply:
invalid